code/datums/bodypart_overlays/mutant_bodypart_overlay.dm

```diff
@@ -13,6 +13,32 @@
 	///Take on the dna/preference from whoever we're gonna be inserted in
 	var/imprint_on_next_insertion = TRUE
 
+/datum/bodypart_overlay/mutant/New(obj/item/organ/attached_organ)
+	. = ..()
+
+	RegisterSignal(attached_organ, COMSIG_ORGAN_IMPLANTED, PROC_REF(on_mob_insert))
+
+/datum/bodypart_overlay/mutant/proc/on_mob_insert(obj/item/organ/parent, mob/living/carbon/receiver)
+	SIGNAL_HANDLER
+
+	if(!should_visual_organ_apply_to(parent.type, receiver))
+		stack_trace("adding a [parent.type] to a [receiver.type] when it shouldn't be!")
+
+	if(imprint_on_next_insertion) //We only want this set *once*
+		var/feature_name = receiver.dna.features[feature_key]
+		if (isnull(feature_name))
+		/* SKYRAT EDIT - Customization - ORIGINAL:
+			feature_name = receiver.dna.species.mutant_organs[parent.type]
+		*/ // SKYRAT EDIT START
+			if(!set_appearance_from_dna(receiver.dna))
+				set_appearance_from_name(receiver.dna.species.mutant_organs[parent.type] || pick(get_global_feature_list()))
+		// SKYRAT EDIT END
+		// SKYRAT EDIT START - Puts the following line in an else block
+		else
+			set_appearance_from_name(feature_name)
+		// SKYRAT EDIT END
```

Comment:
I dont see why this needs to be split so much, can be just one edit start and one edit end comment.

```suggestion
		/* SKYRAT EDIT - Customization - ORIGINAL:
			feature_name = receiver.dna.species.mutant_organs[parent.type]
		set_appearance_from_name(feature_name)
		*/ // SKYRAT EDIT START
			if(!set_appearance_from_dna(receiver.dna))
				set_appearance_from_name(receiver.dna.species.mutant_organs[parent.type] || pick(get_global_feature_list()))
		else
			set_appearance_from_name(feature_name)
		// SKYRAT EDIT END
```